src/chainparams.cpp

```diff
@@ -353,33 +351,78 @@ class CRegTestParams : public CChainParams {
         /* enable fallback fee on regtest */
         m_fallback_fee_enabled = true;
     }
+
+    /**
+     * Allows modifying the Version Bits regtest parameters.
+     */
+    void UpdateVersionBitsParameters(Consensus::DeploymentPos d, int64_t nStartTime, int64_t nTimeout)
+    {
+        consensus.vDeployments[d].nStartTime = nStartTime;
+        consensus.vDeployments[d].nTimeout = nTimeout;
+    }
+    void UpdateVersionBitsParametersFromArgs(ArgsManager& args);
 };
 
+void CRegTestParams::UpdateVersionBitsParametersFromArgs(ArgsManager& args)
+{
+    // Allow overriding version bits parameters for testing
+    if (args.IsArgSet("-vbparams")) {
+        for (const std::string& strDeployment : args.GetArgs("-vbparams")) {
+            std::vector<std::string> vDeploymentParams;
+            boost::split(vDeploymentParams, strDeployment, boost::is_any_of(":"));
+            if (vDeploymentParams.size() != 3) {
+                throw std::runtime_error("Version bits parameters malformed, expecting deployment:start:end");
+            }
+            int64_t nStartTime, nTimeout;
+            if (!ParseInt64(vDeploymentParams[1], &nStartTime)) {
+                throw std::runtime_error(strprintf("Invalid nStartTime (%s)", vDeploymentParams[1]));
+            }
+            if (!ParseInt64(vDeploymentParams[2], &nTimeout)) {
+                throw std::runtime_error(strprintf("Invalid nTimeout (%s)", vDeploymentParams[2]));
+            }
+            bool found = false;
+            for (int j=0; j<(int)Consensus::MAX_VERSION_BITS_DEPLOYMENTS; ++j)
+            {
+                if (vDeploymentParams[0].compare(VersionBitsDeploymentInfo[j].name) == 0) {
+                    UpdateVersionBitsParameters(Consensus::DeploymentPos(j), nStartTime, nTimeout);
+                    found = true;
+                    LogPrintf("Setting version bits activation parameters for %s to start=%ld, timeout=%ld\n", vDeploymentParams[0], nStartTime, nTimeout);
+                    break;
+                }
+            }
+            if (!found) {
+                throw std::runtime_error(strprintf("Invalid deployment (%s)", vDeploymentParams[0]));
+            }
+        }
+    }
+}
+
 static std::unique_ptr<CChainParams> globalChainParams;
```

Comment:
Could change this to `unique_ptr<const CChainParams>` to better ensure that params aren't changed after initialisation? (Means `CreateChainParams()` return value also needs to change, but that seems to work fine)